### ISSUE TITLE
If rollermine is held by player, other rollermines, if they're friendly, attacks held rollermine

### ISSUE THREAD

**Issue by LordofCreepers** [Facepunch/garrysmod-issues#4762]
Yeah, I have already posted similar issue and was too lazy to add anything that would support it. But now I'm back

Yes, rollermines don't attack each other when you set them to be friendly by AddRelationship or AddEntityRelationship. Until you grab one of them with gravgun

Steps to reproduce:
	1. Create Lua file in autorun/server with content:
	--
	hook.Add( "OnEntityCreated", "SetRollerminesFriendly", function( ent )
		if ent:GetClass() ~= "npc_rollermine" then return end
		ent:AddRelationship( "player D_LI 99" )
	end )
	--
        and bootup/reload server
	or (if server is up) open console and enter: lua_run hook.Add( "OnEntityCreated", "SetRollerminesFriendly", function( ent )
		if ent:GetClass() ~= "npc_rollermine" then return end
		ent:AddRelationship( "player D_LI 99" )
	end )
	2. Spawn two rollermines
	3. Grab one of them with gravgun

Personal test report:

[2020-12-08 18-35-23.zip](https://github.com/Facepunch/garrysmod-issues/files/5659999/2020-12-08.18-35-23.zip)

**Comment by Combine900** [comment]
@robotboy655 still a thing? As they are added for the next update.

**Comment by robotboy655** [comment]
Yes, it is still a thing, and it will always be a thing. When you pick up a rollermine, it becomes a friendly one, so obviously, enemies attack it (All combine enemies), and friendlies do not. (Alyx)

This is made to enable the rollermine hacking in Episode 1, where you bring a rollermine to Alyx to hack, so she hacks it instead of shooting it. You do not need any Lua code to have this happen. It will happen with Ignore Players set to 1 as well.

**Comment by Combine900** [comment]
nvm, thanks for the info. Might close this thread :)
